static/js/screens/results.js

```diff
@@ -6,258 +6,340 @@
+
+            api.onResponse("results", (response) => {
+                if (!this.loaded)
+                    this.screen.overlay({ show: false });
+
+                // Cache user info for applying in the future
+                localStorage.setItem("cache.account", JSON.stringify({
+                    email: core.account.email,
+                    name: response.info.name,
+                    studentID: response.info.studentID,
+                    birthday: response.info.birthday,
+                    classroom: response.info.classroom,
+                    department: response.info.department,
+                    tForm: response.info.tForm,
+                }));
+
+                this.loaded = true;
+                emptyNode(this.view.table.tbody);
+                this.screen.set({ subTitle: response.info.mode });
+
+                this.view.info.cpa.value.innerText = response.info.cpa.toFixed(3);
+                this.view.info.grade.value.innerText = response.info.grade;
+
+                for (let item of response.info.results)
+                    this.addListItem(item);
+
+                this.screen.loading = false;
+            });
+        },
+
+        reset() {
+            this.loaded = false;
+            emptyNode(this.view.table.tbody);
+            this.view.info.cpa.value.innerText = "---";
+            this.view.info.grade.value.innerText = "---";
+            this.screen.set({ subTitle: "" });
+        },
+
+        onLogout() {
+            this.reset();
+            this.screen.overlay({
+                icon: "exclamation",
+                title: "Bạn Chưa Đăng Nhập",
+                description: `Hãy đăng nhập vào CTMS để xem nội dung này!`,
+                buttons: {
+                    login: { text: "ĐĂNG NHẬP", icon: "signin", onClick: () => core.account.clickable.active = true }
+                }
+            });
+
+            this.screen.loading = false;
+        },
+
+        async load() {
+            if (!core.account.loggedIn) {
+                this.onLogout();
+                return;
+            }
+
+            try {
+                this.screen.loading = true;
+                await api.results();
+                this.screen.loading = false;
+            } catch (e) {
+                let error = parseException(e);
+
+                this.reset();
+                this.screen.overlay({
+                    icon: "bomb",
+                    title: "Toang Rồi Ông Giáo Ạ!",
+                    description: `<pre class="break">[${error.code}] >>> ${error.description}</pre>`,
+                    buttons: {
+                        login: { text: "THỬ LẠI", color: "pink", icon: "reload", onClick: () => this.load() }
+                    }
+                });
+
+                this.screen.loading = false;
+            }
+        },
+
+
+
+        /**
+         * 
+         * @param	{*}			param0 
+         * @returns {String[]}	array day of 2 week // format :2021-12-06
+         */
+
```

Comment:
Remove this empty line to make the jsDoc belong to the function below.

Reply:
hummm